### PR TITLE
check_drivesize : continue working if drive is not found, so that empty-state can override the output

### PR DESCRIPTION
**Pull request by inqrphl**
- add error types to the helper functions of check_drivesize
- use error types and typecasting to see if no drives are found
- return OK state if no drives are found
- write non-existent drive tests for check_drivesize
- also test customizing empty state result using empty-state= argument

### CODE REVIEW

**Comment by sni** [comment]
> Is this behavior expected from Windows as well?

Yes please, this check should behave the same way across all supported platforms.

File: pkg/snclient/check_drivesize.go [hunk 1]
```diff
@@ -104,7 +105,7 @@ func (l *CheckDrivesize) Build() *CheckData {
 		okSyntax:        "%(status) - All %(count) drive(s) are ok",
 		detailSyntax:    "%(drive_or_name) %(used)/%(size) (%(used_pct | fmt=%.1f )%)",
 		topSyntax:       "%(status) - ${problem_list}",
-		emptyState:      CheckExitUnknown,
```
[the remark this answers is not in the document]
Comment (inqrphl) [reply pkg/snclient/check_drivesize.go]: Hmm ok. I thought you wanted to return OK by default, even if empty-state was unset, and have the user report it as WARNING with empty-state if needed.

Ill revert that line, and fix the doc entry and tests.

File: pkg/snclient/check_drivesize.go [hunk 2]
```diff
@@ -164,12 +165,28 @@ func (l *CheckDrivesize) Check(ctx context.Context, snc *Agent, check *CheckData
 	}
 	requiredDisks := map[string]map[string]string{}
 	drives, err := l.getRequiredDisks(l.drives, false)
-	if err != nil {
+	var notFoundErr *PartitionNotFoundError
+	var notMountedErr *PartitionNotMountedError
+	var partitionDiscoveryErr *PartitionDiscoveryError
+	switch {
+	case errors.As(err, &notFoundErr):
+		log.Debugf("check_drivesize, drive is not found : %s, continuing check with empty disks", notFoundErr.Error())
```
Comment (sni) [inline pkg/snclient/check_drivesize.go]: i think we still should return here, because `drives` is probably nil?

Comment (inqrphl) [reply pkg/snclient/check_drivesize.go]: The requested change was to let the program continue running without populating the drives array, so that the empty-state override can work properly?

```
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd
UNKNOWN - No drives found
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd empty-state=OK
OK - No drives found
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd empty-state=CRIT
CRITICAL - No drives found
```

If I return nil and the notFoundError here it is going to report UNKNOWN even if i override the empty-state

```
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd
UNKNOWN - partition not found for path: /asdasd , error: stat /asdasd: no such file or directory
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd empty-state=OK
UNKNOWN - partition not found for path: /asdasd , error: stat /asdasd: no such file or directory
OMD[demosite@debian13vm]:~$ ./lib/monitoring-plugins/check_nsc_web check_drivesize drive=/asdasd empty-state=CRIT
UNKNOWN - partition not found for path: /asdasd , error: stat /asdasd: no such file or directory
```
